### ISSUE TITLE
[1630] Add overnight background fetch to Pocket channels

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#2188]
UX requirements from https://github.com/mozilla-mobile/firefox-tv/issues/2123#issuecomment-488500101. This issue depends on #2186. This issue is a nice to have for Pocket channels.

### Why/User Benefit/User Problem
After #2186, we are expected to update the Pocket channels every 45 minutes while the app is open. Unfortunately, this means if the app is closed, when users open the app for the first time today, they may see stale content, potentially preventing them from seeing new content and thus they may think Pocket is uninteresting.

### What / Requirements
- [ ] Add an update, even if the app is closed, between maybe 3-5am every day
  - We want to fetch updates when the user is most likely asleep (like Windows Updates)
  - [ ] Ensure there is randomness in scheduling (write a test!): if all devices hit the server at the same time, it won't be good
  - [ ] Ensure there is an exponential back-off that also has randomness (write a test!): see above
  - Recommendation: consider WorkManager & periodic jobs
- [ ] (optional code clean-up) Use this code for the foreground update
  - We can use WorkManager here too because we don't need to immediately update the foreground UI with our fetch; WorkManager handles scheduling and back-off (but does it have randomness?) so it's much simpler than the code we have now
  - [ ] ^ Ensure randomness & exponential back-off
  - Recommended algorithm:
    - When the app is foregrounded (onStart ?):
        - If there is no update scheduled, schedule one 45 minutes from now
        - If there is an update scheduled, do nothing
    - When the update runs:
        - If the app is foregrounded, schedule another update 45 minutes from now
        - If the app is backgrounded, do nothing

### Acceptance Criteria (how do I know when I’m done?)

- Determine the increased cost of pinging server every night for every installed user is acceptable
- The list of pocket recommended videos is updated every night for each user
- Updates do not occur at the same time for all users, to avoid overloading
- Pocket is aware of the change and is prepared to monitor once we launch
  - Tell them our current behavior and how this is different

**Comment by athomasmoz** [comment]
Closing in favour of tracking under #2223
